### DISCUSSION TITLE
Closed source project security

### DISCUSSION THREAD

**Discussion by Kalisnoir** [intitni/CopilotForXcode#211] [General]
I wanna start with I’ve tried this on my personal Mac for a personal project and it’s great, good work! I am wanting to integrate this into my work life too but want to alleviate a concern.
I know that both copilot and openAI via the API are safe for closed source projects, but just wanted to double check there isn’t anything else in this that compromise that? I ask because in the readme it mentions for company projects you can disable suggestions.

**Comment by intitni** [comment]
The feature is there because some companies don't believe that GitHub Copilot is safe for closed source projects. The app won't send network requests to my own server. It will only send requests to:

- GitHub, for GitHub Copilot, update check, download language servers.
- OpenAI, for ChatGPT.
- Any server you set in the field "ChatGPT Server".
- And soon Codeium, and other services to be added in the future.

**Reply by Kalisnoir** [reply]
Thanks for the response @intitni for now I’ll stick to using it on personal projects only to be safe until advised otherwise by my company (by this I mean due to lack of trust for copilot etc, not yourself)
